### PR TITLE
Fix lightning strikes

### LINKED ISSUE #2842: Scripted Lightning Flash brightens level permanently
The lightning flash works fine normally, however, if you disable to lighning object in your level and have it perform a lighnting flash through scripting, using `[Thunderstorm].lightning();` the level begins to slowly brighten up with every strike until the whole screen is just white.

Another thing, it appears that when the background tints it also reduces the alpha value, which should not happen!

### PR DESCRIPTION
**Pull request by weluvgoatz**
Fixes #2842.

### CODE REVIEW

**Comment by Rusty-Box** [comment]
It's not quite fixed. If `running = false`, in the thunderstorm object's properties, and you run the script, the flash is now completely turned off.

Also regarding the background darkening, while the transparency issue seems fixed, now instead of remaining darkened until lightning strikes it gets darker for but a short moment.

**Comment by weluvgoatz** [comment]
> It's not quite fixed. If `running = false`, in the thunderstorm object's properties, and you run the script, the flash is now completely turned off.

Well that's how it is currently in the nightly.  I don't recall you specifying this issue in the original report.  Did you want some other behavior here instead?

> Also regarding the background darkening, while the transparency issue seems fixed, now instead of remaining darkened until lightning strikes it gets darker for but a short moment.

Is the intended behavior to stay dark? If so I can change that.  I didn't know.

**Comment by Rusty-Box** [comment]
> Well that's how it is currently in the nightly. I don't recall you specifying this issue in the original report. Did you want some other behavior here instead?

I wasn't aware this made a difference. If you ask me it should work no matter what as the lightning script causes lightning to happen no matter what. So the flash should work in all cases!

> Is the intended behavior to stay dark? If so I can change that. I didn't know.

It is indeed intended. People wanted a visual indicator for when lightning is about to strike in case they play with no audio.

**Review by Rusty-Box — APPROVED** [review]
Seems to be working!

**Review by Frostwithasideofsalt — APPROVED** [review]
Tested it, seems to work

File: src/object/thunderstorm.hpp [hunk 1]
```diff
@@ -60,7 +61,7 @@ class Thunderstorm final : public GameObject,
   void thunder();
 
   /** Play lightning, i.e. call flash() and electrify() */
-  void lightning();
+  void lightning(bool is_scripted = false);
```
Comment (Vankata453) [inline src/object/thunderstorm.hpp:64]: When we migrate to `simplesquirrel`, the `bool is_scripted` parameter wouldn't be supported - the scripting function to be exposed should rather be a separate function in the code, preferably with the same name as the one it's exposed with (easier for documentation).

**Pull request merged**